docs/en/faq/why-kotlin-jdsl-returns-nullable-list-in-the-spring-support-module.md

```diff
@@ -0,0 +1,39 @@
+# Why does Kotlin JDSL allow nullable return types?
+
+Kotlin JDSL helps developers develop more efficiently by allowing them to write queries without creating a separate metadata model or making string syntax errors.
+However, Kotlin JDSL users may sometimes have questions about nullable return types in the form of Slice<T?> or Page<T?> returned when calling APIs like findSlice or findPage methods.
+
+## Why do we need nullable return types?
+
+The reason why Kotlin JDSL supports nullable return types is that depending on how the JPQL query is written, some of the returned list items may contain null values.
+The simplest example is when you look up a column or entity without using DTO Projection.
+
+The table below shows some of the cases where nullable return types can occur when using the Kotlin JDSL.
+
+| Item           | Nullable or not | Reason                                                                                               |
+|----------------|-----------------|------------------------------------------------------------------------------------------------------|
+| DTO Projection | X               | Calling the constructor for all ROWs results in the object being created, which does not allow nulls |
```

Comment:
This is inaccurate information. In case of DTO Projection, Nullable is also allowed. For example

```kotlin
class Entity {
   @Id
   var id: Long? = null

   @Column(nullable = true)
   var name: String? = null
}
```
When there is an entity like the above, null may occur at the time of the query below.
select name from Entity;

Reply:
This is correct information. When using the new keyword in JPQL, the constructor is called for every row to create the object. That's why there can't be a nullable for DTO Projection only.

![image](https://github.com/line/kotlin-jdsl/assets/31586979/78cf64b3-dbd3-4e64-8909-7394bd79348c)
![image](https://github.com/line/kotlin-jdsl/assets/31586979/9a9ce1ba-2a2f-4a77-8eee-1839ba6643ba)

Reply:
If you mean returning a DTO object itself, that is correct.
I meant that the fields in the object are nullable, but that seems to be incorrect information. Thanks for the good point.

Reply:
Ah, I see you got it, I'll add it to the documentation!

Reply:
It's nice to be able to check it with test code.

![image](https://github.com/line/kotlin-jdsl/assets/66164361/5022ee2a-ab9d-4dc0-9f4e-d855c074289d)